bundles/org.openhab.binding.meteoalerte/src/main/java/org/openhab/binding/meteoalerte/internal/MeteoAlertIconProvider.java

```diff
@@ -0,0 +1,111 @@
+ */
+package org.openhab.binding.meteoalerte.internal;
+
+import static org.openhab.binding.meteoalerte.internal.MeteoAlerteBindingConstants.*;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.Set;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.meteoalerte.internal.json.ResponseFieldDTO.AlertLevel;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.types.State;
+import org.openhab.core.ui.icon.IconProvider;
+import org.openhab.core.ui.icon.IconSet;
+import org.openhab.core.ui.icon.IconSet.Format;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link MeteoAlertIconProvider} is the class providing binding related icons.
+ *
+ * @author Gaël L'hopital - Initial contribution
+ */
+@Component(immediate = true, service = { IconProvider.class })
+@NonNullByDefault
+public class MeteoAlertIconProvider implements IconProvider {
+    private static final Set<String> ICONS = Set.of(WAVE.replace("-", "_"), AVALANCHE, HEAT, FREEZE.replace("-", "_"),
+            FLOOD, SNOW, STORM, RAIN.replace("-", "_"), WIND);
+    private static final IconSet ICON_SET = new IconSet(BINDING_ID, "Météo Alerte Icons",
+            "These are the icons describing weather events", Set.of(Format.SVG));
+
+    private final Logger logger = LoggerFactory.getLogger(MeteoAlertIconProvider.class);
+    private final ClassLoader classLoader;
+
+    @Activate
+    public MeteoAlertIconProvider() {
+        classLoader = Objects.requireNonNull(MeteoAlertIconProvider.class.getClassLoader());
+    }
+
+    @Override
+    public Set<IconSet> getIconSets() {
+        return Set.of(ICON_SET);
+    }
+
+    @Override
+    public Set<IconSet> getIconSets(@Nullable Locale locale) {
+        return getIconSets();
+    }
+
+    @Override
+    public @Nullable Integer hasIcon(String category, String iconSetId, Format format) {
+        return ICONS.contains(category) && format == Format.SVG ? 7 : null;
+    }
+
+    @Override
+    public @Nullable InputStream getIcon(String category, String iconSetId, @Nullable String value, Format format) {
+        String icon = getResource(category);
+
+        if (icon.isEmpty()) {
+            return null;
+        }
+
+        if (value != null) {
+            try {
+                State state = DecimalType.valueOf(value);
+
+                AlertLevel alertLevel = ALERT_LEVELS.entrySet().stream().filter(entry -> entry.getValue().equals(state))
+                        .findFirst().map(entry -> entry.getKey()).orElse(AlertLevel.UNKNOWN);
+
+                icon = icon.replaceAll(UNKNOWN_COLOR, ALERT_COLORS.getOrDefault(alertLevel, UNKNOWN_COLOR));
```

Comment:
Ok, I understand why you did not use `AbstractResourceIconProvider`, you implemented a specific way to retrieve the icon based on the item state, without storing several different icons, as the only difference is the colour.

Reply:
You're correct. `AbstractResourceIconProvider` is not usefull for me here.